### DISCUSSION TITLE
What file extension to use?

### DISCUSSION THREAD

**Discussion by NicolasCARPi** [TheELNConsortium/TheELNFileFormat#2] [Polls]
What do you think the file extension should be?

On one hand we have `.eln` which is unknown to the world, and makes the archive an unknow binary file. Advanced users know that they can open it as if it was any other zip, the same way one can open a .docx file. Keeping this extension might prevent users from fiddling with it, zipping it back again and we end up with an incorrect archive. The idea being that the archive should be immutable, and inspection of content by a human should only be done by devs. The end users don't really have to know what is the internal structure.

On the other hand, we have `.eln.zip`, the same way we have `.tar.gz` which is a gzipped tarball, we have a zipped eln/ro-crate. The advantages are that it makes it easy to be opened to inspect content on the desktop, because it is now obvious that it is a zip, it also means that unarchivers will be able to open it easily and OS actions (right click > unzip) will be available.

Both options are valid with their pros and cons.

**Comment by NicolasCARPi** [comment]
Pinging @bronger @CEschke @FlorianRhiem @martin-held-hereon @mikee1976 @nicobrandt @nicolejung @SteffenBrinckmann (btw you should subscribe to activity on this repo!)

**Comment by bronger** [comment]
Sorry, could not resist: https://wilson.bronger.org/12246653_789997134444585_8425954147094342405_n.jpg

**Reply by NicolasCARPi** [reply]
> There are only two hard things in Computer Science: cache invalidation and naming things.

Phil Karlton

;)

**Comment by nicobrandt** [comment]
I feel like simply using `.eln` is clearer, as `.eln.zip` would imply that there is an actual `.eln` file inside the ZIP archive. Unless that is the idea, but then the problem is just shifted by one layer ;) I also like the arguments that you mention for using `.eln` more, and as long as people know it is actually a regular ZIP archive, it should be fine (many archiving tools/previewers actually don't care about the file extension anways).

Alternatively, we could also just use the regular `.zip` extension. Of course, this would require looking at the contents of the archive to determine whether we are dealing with an "ELN" file or not. At the same time, the file extension does not guarantee that the content is according to the standard, either.

**Reply by bronger** [reply]
Moreover, LibreOffice files or ePub documents also are ZIP, but with a simple non-ZIP file extension.

**Reply by NicolasCARPi** [reply]
Yes, yesterday I was more encline to `.eln.zip` but today I'm rooting for `.eln`!

**Comment by NicolasCARPi** [comment]
If I change my vote to `.eln` then it wins the poll. So let's stick with that then.

**Comment by mikee1976** [comment]
+1 :-D

Viele Grüße
Michael
